scripts/validation-suite.sh

```
@@ -0,0 +1,377 @@
+            METRICS_STATUS="PASS"
+            log_success "Metrics validation: PASSED"
+        else
+            METRICS_STATUS="FAIL"
+            log_fail "Metrics validation: FAILED"
+        fi
+
+        # If baseline provided, compare metrics
+        if [[ -n "$BASELINE_FILE" && -f "$BASELINE_FILE" ]]; then
+            log_info ""
+            log_info "Running metrics comparison against baseline..."
+            if ! "$SCRIPT_DIR/compare-metrics.sh" $COMMON_ARGS -c "$BASELINE_FILE" > /dev/null; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)